### ISSUE TITLE
When using new cache ability getting a console warning regarding circular dependency

### ISSUE THREAD

**Issue by omerman** [vercel/swr#327]
First of all, everything works fine and as expected :)..

I'm just getting a  warning that I guess should be addressed

![image](https://user-images.githubusercontent.com/6780529/78301602-03302680-7542-11ea-9b19-e2addb290683.png)

Steps to reproduce -> create a fresh RN app (non expo for that matter..)
```
import useSwr, {cache} from 'swr'; // import and off course use them both so it won't get tree shaked.
```

I'll also try and take a crack at it soon.

Cheers

**Comment by henry-ns** [comment]
I got this warning too, using SWR on react-native

![Screenshot from 2020-07-16 13-31-13](https://user-images.githubusercontent.com/16365204/87697812-fdfd8d00-c768-11ea-8c49-72870fc7377a.png)

**Comment by huozhi** [comment]
@omerman @henry-ns  thanks for reporting! this issue was been resolved in #493 . the change will be delivered into next release.

**Comment by shuding** [comment]
Closing as we've landed a new Cache API in [SWR 1.0](https://swr.vercel.app/blog/swr-v1), please check it out!
